bootcompiler/src/main/scala/org/mozartoz/bootcompiler/transform/Desugar.scala

```diff
@@ -198,6 +220,14 @@ object Desugar extends Transformer with TreeDSL {
       treeCopy.MatchStatementClause(clause, pattern, guard, exprToBindStatement(result, body))
   }
 
+  def hasSideEffects(node: Node): Boolean = node match {
+    case Variable(sym)               => false
+    case Constant(value)             => false
+    case RecordField(feature, value) => hasSideEffects(feature) || hasSideEffects(value)
+    case Record(label, fields)       => hasSideEffects(label) || fields.exists(hasSideEffects(_))
```

Comment:
Normally, it should not happen for record features or record labels to have side effects, but I thought it was actually safer to do so.